app/models/manageiq/providers/ibm_cloud/object_storage/storage_manager.rb

```diff
@@ -212,4 +212,14 @@ def remove_object(bucket_name, object_name)
   def queue_name_for_ems_refresh
     queue_name
   end
+
+  def required_credential_fields(type)
+    if type == 'default'
+      [:auth_key]
+    elsif type == 'bearer'
+      [:password]
+    else
+      []
+    end
```

Comment:
Best to `.to_s` the type in case e.g. `:default` is passed in

Also up to you but could be a good use for case/when e.g.
```suggestion
    case type.to_s
    when "default" [:auth_key]
    when "bearer"  [:password]
    else           []
    end
```

Reply:
Also for `bearer` it looks like you need `%i[userid password]`